src/shared/debug/launchConfiguration.ts

```diff
@@ -0,0 +1,64 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as _ from 'lodash'
+import * as vscode from 'vscode'
+import { AwsSamDebuggerConfiguration } from '../sam/debugger/awsSamDebugConfiguration'
+import { AWS_SAM_DEBUG_TYPE } from '../sam/debugger/awsSamDebugger'
+
+/**
+ * Reads and writes DebugConfigurations.
+ */
+export interface DebugConfigurationSource {
+    getDebugConfigurations(): vscode.DebugConfiguration[]
+    updateDebugConfigurations(value: vscode.DebugConfiguration[]): Promise<void>
+}
+
+/**
+ * Wraps read and write operations on launch.json.
+ */
+export class LaunchConfiguration {
+    /**
+     * Creates a Launch Configuration scoped to the given resource.
+     */
+    public constructor(
+        resource: vscode.Uri,
+        private readonly configSource: DebugConfigurationSource = new DefaultDebugConfigSource(resource)
+    ) {}
+
+    public getDebugConfigurations(): vscode.DebugConfiguration[] {
+        return this.configSource.getDebugConfigurations()
+    }
+
+    public getSamDebugConfigurations(): AwsSamDebuggerConfiguration[] {
+        return _(this.getDebugConfigurations())
+            .filter({ type: AWS_SAM_DEBUG_TYPE })
+            .map(config => config as AwsSamDebuggerConfiguration)
+            .value()
```

Comment:
is lodash really required for map/filter?

Reply:
Required: no, Useful: absolutely

Lodash gives the `.filter({ type: AWS_SAM_DEBUG_TYPE })` syntax which I think reads better. It's *much* more powerful in its pattern matching and that just touches the surface.

js normally requires a predicate function. I think we should consider lodash an extension of the stdlib and use it whenever it's nicer.

Reply:
It would be great if we could remove lodash, so we have one less dependency to manage (think vulnerability updates, etc). I'm not familiar with it though, so there may be some great benefits we get from it that I am not aware of.

It took me a while to read the lodash related calls (due to low familiarity with the library). I was also wondering if more pure map related calls would be sufficient. Collapsing syntax to a minimal form can backfire if it increases time to comprehend code.

Reply:
Yeah, I've read through the lodash doc pretty thoroughly and I'm generally familiar with what the js stdlib (if you can call it that) offers.

The js lib is extremely barebones. Lodash can produce significantly easier to follow (/ write) and less error-prone code IMO.

A very basic example is `_(myArray).isEmpty()`. This is such a basic convenience that it's crazy that js requires `implicit falsiness of myArray.length or myArray.length === 0`. I can't think of another OOP language that doesn't have an `isEmpty` method 😐

Reply:
I think the main reason why we wanted to do away with lodash in the first place is because we were pretty much using it exclusively for HTML templates. I know the EventBridge Schemas are using lodash for debounces, so there's definitely utility in keeping it around, especially if we can expand its usage (so we don't wonder again why we're using lodash for a couple cases)